sphinxlint.py

```diff
@@ -236,6 +235,20 @@ def check_missing_space_after_literal(file, lines, options=None):
                 )
 
 
+def escape2null(text):
+    """Return a string with escape-backslashes converted to nulls."""
+    parts = []
+    start = 0
+    while True:
+        found = text.find('\\', start)
+        if found == -1:
+            parts.append(text[start:])
+            return ''.join(parts)
+        parts.append(text[start:found])
+        parts.append('\x00' + text[found+1:found+2])
+        start = found + 2               # skip character after escape
```

Comment:
What's the purpose of this function?
Can't `\\` used directly in the regex below?
Or can't a simple `text.replace('``', '\x00')` used to implement this?

Reply:
I added a comment to explain why it is usefull. The function is not replacing backticks but backslahes, to tell appart real backslashes and escaping backslashes, here's a copy of the added comment:

    It ease telling appart escaping-backslashes and normal backslashes
    in regex.

    For example : \\\\\\` is hard to match, even with the eyes, it's
    hard to know which backslash escapes which backslash, and it's
    very hard to know if the backtick is escaped.

    By replacing the escaping backslashes with another character they
    become easy to spot:

    0\0\0\`

    (This example uses zeros for readability but the function actually
    uses null bytes, \x00.)

    So we easily see that the backtick is **not** escaped: it's
    preceded by a backslash, not an escaping backslash.